### ISSUE TITLE
minetest.conf mapgen flags for games are ignored

### ISSUE THREAD

**Issue by ensayia** [minetest/minetest#12829]
##### Minetest version

```
5.6.0
```

##### OS / Hardware

Operating system: Arch Linux
CPU: Intel i5-9660

##### Summary

Map generation flags in minetest.conf inside my custom game folder are being ignored. I can verify this my comparing the map_meta.txt of a world I generate and manually disable caves and dungeons by unchecking the boxes at creation with the map_meta.txt of a world where the minetest.conf in my game folder sets those values.

I've looked over the documentation on the Mintest API, the Mintest Modding Book, searched the forums, searched the issue tracker, and reached out to individuals on IRC all with no clues as to a solution or known bug.

my full minetest.conf:
```
naturalslopeslib_enable_surface_update = false
naturalslopeslib_enable_shape_on_dig_place = false

enable_pvp = false
pitch_move = true
fast_move = false # handled manually in the code instead

movement_speed_walk   = 4
movement_speed_crouch = 4
movement_acceleration_default = 4

movement_gravity = 12.75 # 9.81*1.3
movement_speed_jump = 8 # default is 6.5, higher to offset stronger gravity

dedicated_server_step = 0.05 #tick rate

mg_flags = nocaves,nodungeons,light,decorations,biomes
mgv7_spflags = mountains, ridges, nofloatlands, nocaverns 
```
relavent map_meta.txt when using minetest.conf settings:
```
...
mgv7_large_cave_num_min = 0
mg_flags = caves, dungeons, light, decorations, biomes, ores
mg_name = v7
mgv7_small_cave_num_max = 0
mgv7_large_cave_num_max = 2
mgv7_floatland_ymin = 1024
chunksize = 5
water_level = 1
mapgen_limit = 31007
...
```
relevant map_meta.txt when setting the desired map features via checkbox at world creation:
```
...
mgv7_large_cave_num_min = 0
mg_flags = nocaves, nodungeons, light, decorations, biomes, ores
mg_name = v7
mgv7_small_cave_num_max = 0
mgv7_large_cave_num_max = 2
mgv7_floatland_ymin = 1024
chunksize = 5
water_level = 1
mapgen_limit = 31007
...
```

**Comment by SmallJoker** [comment]
1. Flags are processed according to the configuration file hierarchy: https://github.com/minetest/minetest/blob/master/src/settings.cpp#L575
2. The hierarchy looks as follows: https://github.com/minetest/minetest/blob/master/src/settings.h#L65 (where map is after `SL_GLOBAL`).

So if you have no `mg_flags` defined in your global `minetest.conf`, the world should use the values from defaults -> game directly when created from CLI. Note that the main menu GUI might re-create this setting.

**Comment by ensayia** [comment]
I can verify there are no defaults being set in my global minetest.conf. The entirety of it is here:
```
update_last_known = 
screen_h = 2528
screen_w = 3840
update_last_checked = 1664915494
menu_last_game = radiant
server_announce = false
enable_damage = true
creative_mode = false
selected_world_path = /home/ensayia/.minetest/worlds/world8
mainmenu_last_selected_world = 1
maintab_LAST = local
secure.trusted_mods = formspec_edit
fast_move = true
free_move = true
noclip = true
```
I can check in the settings dialog and see that the global game settings have all those flags enabled, but shouldn't my own game's minetest.conf flags override those? What would even be the point of letting the game developer specify what is allowed and not allowed on map generation if Minetest defaults to allowing everything, and overrides a game/mod's configuration?

I'm quite confused.

**Comment by SmallJoker** [comment]
> but shouldn't my own game's minetest.conf flags override those?

Yes. The game's `minetest.conf` file *should* overwrite the default checkbox state for the main menu but let the user select their preferences. To my knowledge this does yet not happen. <--- this is probably the feature request (or bugfix) you want.

FYI: The underlying code is still the same as when I fixed the mapgen flags' functionality back in #10318, hence with an adjustment of the Settings API to retrieve the game settings, it should be easy to implement this.
